### DISCUSSION TITLE
Connecting to Mesh Api : PEM Certificate

### DISCUSSION THREAD

**Discussion by samurech** [Volue-Public/energy-mesh-python#144] [Q&A] [answered]
Hi 
Since we installed the latest mesh api package from github we are not able to connect to our mesh service anymore. Last week everything worked fine and we could query data. Using the same line of code as last week we get this error message when executing "connection.get_version()": 
![image](https://user-images.githubusercontent.com/90327705/156207363-ac5a4279-ac51-4308-8435-bdf2c8a9bae4.png)

We are not sure why, but we realized then a difference in your quickstart example: the connection function is not using the parameter "secure_connection" anymore but requires a "root_pem_certificate". 
![image](https://user-images.githubusercontent.com/90327705/156208038-96257993-1020-4373-9a27-d95ea5758ad2.png)
![image](https://user-images.githubusercontent.com/90327705/156208277-df5052a7-d406-4f03-bf49-09c207407053.png)

Could this be the reason? And I have absolutely no idea what that is. Could you please explain how this should work now?

Thanks for your feedback!

Best regards
Samuel Rechsteiner
Axpo

**Comment by tnoczyns-volue** [comment, marked as answer]
Hi,

Most likely you are using Mesh in version <= 2.2. Could you check what is the version of Mesh you are trying to connect to?
Current master requires Mesh v2.3 which should be available externally starting end of this week.
Meanwhile for Mesh 2.2 please use this:
https://github.com/PowelAS/sme-mesh-python/tree/Mesh_v2.2

Regarding the `secure_connection` to `root_pem_certificate` change: previously there was a bool parameter `secure_connection` in `Connection` class. When set to True it meant that we requested secure (using TLS  certificate) connection to the server. However the path to TLS certificate was hardcoded. To make it flexible now we removed the hardcoded path and enabled users to provide the TLS certificate (in PEM format - base64 encoded string)  directly when initializing `Connection` class. For example you can load a TLS certificate from the disk and provide as this parameter. Please take a look at the function `_get_connection_info` used in examples:
https://github.com/PowelAS/sme-mesh-python/blob/master/src/volue/mesh/examples/__init__.py

The secure TLS connection is required for authentication, here is more info about it:
https://vigilant-eureka-e3845ca2.pages.github.io/authentication.html

In case something is unclear or you have other questions just let me know.

Kind regards,
Tomasz
